snuba/query/snql/parser.py

```diff
@@ -1039,6 +1039,8 @@ def _treeify_or_and_conditions(
     codebase which assume `or` and `and` have two arguments
 
     Adding this post-process step is easier than changing the rest of the query pipeline
+
+    Note: does not apply to the conditions of a from_clause subquery (the nested one)
```

Comment:
wait what? How?

Reply:
see test_treeify_or_and_conditions.py, test_composite

Reply:
I see that you have written a test for it but I don't understand why this holds true

Reply:
After looking in deeper, I see that `transform_expressions` is not implemented for composite queries. You should explain that in the test that you wrote and maybe here as well. Otherwise someone is going to come to that code later, implement it and not understand why we don't transform composite subquery expressions but assume that it's just supposed to be this way. In reality this is an oversight